_posts/2021-04-07-upgrade-improvements-and-migration.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title:  "Upgrade changes and API versions conversion"
+date: 2021-04-07
+author: jakub_stejskal
+---
+Upgrade process of Strimzi and Kafka is not trivial. 
+In Strimzi 0.22 we introduced several improvements, which will make the process easier for the users. 
+Moreover, we also introduced new API version `v1beta2` which will replace `v1beta1` because [Kubernetes 1.22 will remove support](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122) for `apiextensions/v1beta1`.
+This blog post will show you how the upgrade process changed and what you should do to keep your cluster working without issues.
+
+<!--more-->
+
+### Upgrade improvements
+
+Legacy upgrade process was a little complicated.
+Users had to have fill Kafka `version`, but also `log.message.format.version` or `inter.broker.protocol.version` before the upgrade.
+This makes the upgrades hard to execute fully automatically for example when you use Strimzi installed from [OperatorHub](https://operatorhub.io/operator/strimzi-kafka-operator).
+You also needed to upgrade through all versions of Strimzi and Kafka.
+For example if you wanted to upgrade from 0.18 to 0.20, you had to install 0.19 firstly and then do the same steps for 0.20.
+
+Now, the process is not too strict.
+Strimzi now does the following things regarding Kafka configuration:
+* detect used Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` (from CR or compute it from Kafka `version`)
+* configure Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` if it's not specified by user
+
+The main improvement is, that you don't need to upgrade across all Strimzi versions, but you can upgrade from for instance 0.18 directly to 0.22 as it is shown in the diagram bellow.
+You can face the following states:
+* Supported Kafka `version` and `log.message.format.version` or `inter.broker.protocol.version` is set in CR - Strimzi will keep the values and missing one will se based on `version`.
+* Supported Kafka `version` is set, `log.message.format.version` and `inter.broker.protocol.version` are not set in CR - Strimzi will set `log.message.format.version` and `inter.broker.protocol.version` based on `version`
+* Supported Kafka `version` is not set in CR - Strimzi will set Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` based on the default Kafka supported in 0.22.
+* Unsupported Kafka `version` is set in CR - Strimzi will set `Kafka` CR status to `NotReady`. 
+  In that case user has to update CR and manually set a proper config.
```

Comment:
I think you should go into more details here. Especially with regard to what happens when the values are not set. The important point being that they will be:
* First set to the version from the old Kafka => to make it possible to roll the cluster safely
* Then they will be automatically updated to the new version.

You can also finish it with some _how many restarts to expect_ paragraph.

And TBH, I think that would make more interesting diagram than the one you have ;-)